src/main/java/org/schemaspy/output/dot/schemaspy/DotFile.java

```diff
@@ -0,0 +1,33 @@
+package org.schemaspy.output.dot.schemaspy;
+
+import org.schemaspy.output.dot.DotException;
+import org.schemaspy.output.dot.schemaspy.graph.EmptyGraph;
+import org.schemaspy.output.dot.schemaspy.graph.Graph;
+import org.schemaspy.util.Writers;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.file.Path;
+
+public class DotFile implements DotWriter {
+    private final GraphProducer graphProducer;
+    private final Path file;
+    public DotFile(GraphProducer graphProducer, Path file) {
+        this.graphProducer = graphProducer;
+        this.file = file;
+    }
+
+    public boolean write() {
```

Comment:
[Why](https://en.wikipedia.org/wiki/Command%E2%80%93query_separation) is the return type needed? Is it legacy reasons?

Reply:
Not legacy, but for some reason I wanted OrphanDiagram to be responsible for the filtering.

If it's filtered and nothing returned, an empty Graph is returned and nothing is written.

I'll have another crack at it since the abort or cancelation should be done higher in the object graph.

Reply:
I was stuck in an old misconception. That avoiding stuff is important.

So I'm gonna remove all the checks, and just having it creating a diagram with no nodes if there aren't any.